build.gradle.kts

```diff
@@ -225,12 +227,27 @@ tasks {
 	}
 
 	javadoc {
-		// of all the javadoc checks (accessibility, html, missing, reference, syntax; see
-		// https://docs.oracle.com/javase/8/docs/technotes/tools/unix/javadoc.html#BEJEFABE)
-		// disable the warning for missing comments and tags because they spam the output
-		// (it does often not make sense to comment every tag; e.g. the @return tag on annotations)
-		(options as CoreJavadocOptions).addStringOption("Xdoclint:accessibility,html,syntax,reference", "-quiet")
-		options.encoding = "UTF-8"
+		// Exclude internal implementation package from javadoc
+		exclude("org/junitpioneer/internal")
+
+		options {
+			// Cast to standard doclet options, see https://github.com/gradle/gradle/issues/7038#issuecomment-448294937
+			this as StandardJavadocDocletOptions
+
+			links = listOf("https://junit.org/junit5/docs/current/api/")
+
+			// Set javadoc `--release` flag (affects which warnings and errors are reported)
+			// (Note: Gradle adds one leading '-' to the option on its own)
+			addStringOption("-release", targetJavaVersion.getMajorVersion())
+
+			// Enable doclint, but ignore warnings for missing tags, see
+			// https://docs.oracle.com/en/java/javase/17/docs/specs/man/javadoc.html#additional-options-provided-by-the-standard-doclet
+			// The Gradle option methods are rather misleading, but a boolean `true` value just makes sure the flag
+			// is passed to javadoc, see https://github.com/gradle/gradle/issues/2354
+			addBooleanOption("Xdoclint:all,-missing", true)
```

Comment:
Why `addBooleanOption` instead of `addStringOption`?

Also, the [Javadoc manual](https://docs.oracle.com/en/java/javase/17/docs/specs/man/javadoc.html#additional-options-provided-by-the-standard-doclet) says:

```text
-Xdoclint:(all|none|[-]group)
```

I'm not that familiar with the Javadoc CLI, but does it really make sense to combine `all` and `missing` (especially with a leading `-`)?

```diff
-Xdoclint:accessibility,html,syntax,reference
+Xdoclint:all,-missing
```

Reply:
> Why `addBooleanOption` instead of `addStringOption`?

This is explained a bit in the Gradle issue linked in the comment; the name and behavior of these methods is rather confusing. Apparently Gradle considers a "boolean option" an option without separate value, e.g. `-option`. Whereas a "string option" would be an option with separate value, e.g. `-option value`. Since the doclint setting is part of the same argument it has to be a "boolean option".
However, using a "string option" with `null` value (explicitly or when calling the `addStringOption` method without separate value parameter), the option seems to be simply omitted from the command line arguments (see [Gradle implementation](https://github.com/gradle/gradle/blob/fc26c4c3e3edb3ef3572a4b663fc452689f2ad19/subprojects/language-java/src/main/java/org/gradle/external/javadoc/internal/StringJavadocOptionFileOption.java#L28-L30)).

---

> I'm not that familiar with the Javadoc CLI, but does it really make sense to combine `all` and `missing` (especially with a leading `-`)?

The leading `-` excludes the group. I tried to explain that with the comment "Enable doclint, but ignore warnings for missing tags" a few lines above, but maybe that was not obvious; do you have any suggestions for improving the comment?
This configuration is mentioned by the following bullet point in the linked documentation:
> <code>-Xdoclint all,-<i>group</i></code>: Enables all checks except <i>group</i> checks

Reply:
Thank you for that great explanation!